### PR TITLE
Revendor gardener and MCM

### PR DESCRIPTION
**Pull request by kon-angelo**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement dependency
Revendor gardener to `v1.69.3`
```
```improvement dependency
Revendor MCM to `v0.49.0`
```
```breaking operator
Support for migration of machineClass is dropped by the mcm-provider
```

### CODE REVIEW

File: pkg/driver/driver.go [hunk 1]
```diff
@@ -203,31 +202,3 @@ func (p *OpenstackDriver) GetVolumeIDs(_ context.Context, req *driver.GetVolumeI
 	}
 	return &driver.GetVolumeIDsResponse{VolumeIDs: names}, nil
 }
-
-// GenerateMachineClassForMigration helps in migration of one kind of machineClass CR to another kind.
```
Comment (himanshu-kun) [inline pkg/driver/driver.go:207]: we can also remove the kind `openStackMachineClassKind`

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
a small nit otherwise looks good , hope you have run the integration tests :) 

Also there is a probable error which we might need to fix in mcm v0.49.0 , so kindly don't release.

File: pkg/driver/driver.go (hunk 1 above)
Comment (kon-angelo) [reply pkg/driver/driver.go:207]: Already done, PTAL :)

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**